lib/ToolbarPanelHub.jsm

```diff
@@ -137,23 +137,45 @@ class _ToolbarPanelHub {
     });
   }
 
+  // Newer messages first and use `order` field to decide between messages
+  // with the same timestamp
+  _sortWhatsNewMessages(m1, m2) {
+    // Sort by published_date in descending order.
+    if (m1.content.published_date === m2.content.published_date) {
+      // Ascending order
+      return m1.order - m2.order;
+    }
+    if (m1.content.published_date > m2.content.published_date) {
+      return -1;
+    }
+    return 1;
+  }
+
   // Render what's new messages into the panel.
   async renderMessages(win, doc, containerId) {
-    const messages = (await this.messages).sort((m1, m2) => {
-      // Sort by published_date in descending order.
-      if (m1.content.published_date === m2.content.published_date) {
-        return 0;
-      }
-      if (m1.content.published_date > m2.content.published_date) {
-        return -1;
-      }
-      return 1;
-    });
+    const messages = (await this.messages).sort(this._sortWhatsNewMessages);
     const container = doc.getElementById(containerId);
 
     if (messages && !container.querySelector(".whatsNew-message")) {
       let previousDate = 0;
+      // Get and store any variable part of the message content
+      this.state.contentArguments = await this._contentArguments();
       for (let message of messages) {
+        // Only render date if it is different from the one rendered before.
+        if (message.content.published_date !== previousDate) {
```

Comment:
I see this code was just moved, but I guess we can just be careful to make sure to use the exact same date value down to the millisecond for messages that should share the same date header. Otherwise, we could do some math to truncate to the day.. although not sure if that has issues with time zones and daylight saving… let's just leave this as is 😛

Reply:
I'm counting on the staging environment + QA to catch any issues, we haven't seen any timezone weirdness yet. Probably because it's rounded down to a day not something more precise.